tests/playwright-test/web-server.spec.ts

```diff
@@ -35,13 +35,38 @@ test('should create a server', async ({ runInlineTest }, { workerIndex }) => {
         webServer: {
           command: 'node ${JSON.stringify(path.join(__dirname, 'assets', 'simple-server.js'))} ${port}',
           port: ${port},
-        }
+        },
+        globalSetup: 'globalSetup.ts',
+        globalTeardown: 'globalTeardown.ts',
+      };
+    `,
+    'globalSetup.ts': `
+      module.exports = async () => {
+        console.log('globalSetup')
+        return () => console.log('globalSetup teardown');
+      };
+    `,
+    'globalTeardown.ts': `
+      module.exports = async () => {
+        const http = require("http");
+        const response = await new Promise(resolve => {
+          const request = http.request("http://localhost:${port}/hello", resolve);
+          request.end();
+        })
+        console.log('globalTeardown-status-'+response.statusCode)
       };
     `,
   });
   expect(result.exitCode).toBe(0);
   expect(result.passed).toBe(1);
   expect(result.report.suites[0].specs[0].tests[0].results[0].status).toContain('passed');
+
+  const expectedLogMessages = ['Starting WebServer', 'globalSetup', 'globalSetup teardown', 'globalTeardown-status-200'];
```

Comment:
We usually use `%%` for this kind of stuff: https://github.com/microsoft/playwright/blob/master/tests/playwright-test/fixtures.spec.ts#L619-L625

Reply:
Tried, but since Starting WebServer is quite dynamic (followed by the command which is escaped) I would keep it like that for now.